### PR TITLE
[FIX] chart panel: title state in chart design panel 

### PR DESCRIPTION
**Pull request by jash-odoo**
## Description:

Implemented t-model for state management of the chart title within
the sidepanel. By using t-model, the input value remains persistent
throughout rerendering, even when triggered by dispatches like multi-user
commands. This implementation ensures consistent display of the correct
value entered in the title input

Odoo task ID : [2991474](https://www.odoo.com/web#id=2991474&cids=2&menu_id=4720&action=333&active_id=2328&model=project.task&view_type=form)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/side_panel/chart/line_bar_pie_panel/design_panel.ts [hunk 1]
```diff
@@ -44,9 +48,9 @@ export class LineBarPieDesignPanel extends Component<Props, SpreadsheetChildEnv>
     });
   }
 
-  updateTitle(ev) {
+  updateTitle() {
     this.props.updateChart(this.props.figureId, {
-      title: ev.target.value,
+      title: _t(this.state.title),
```
Comment (hokolomopo) [inline src/components/side_panel/chart/line_bar_pie_panel/design_panel.ts:53]: The translation should be at the setup, not here. 

The title of the chart might be translated (we extract chart titles for translation in dashboard), so translatting it at startup make sense. But the new title that the user just inputted will not be in the translation and even if it was a translatable string, we want to store the non-translated one in DB.

**Review by hokolomopo — COMMENTED** [review]
Seems good :+1:

**Review by rrahir — CHANGES_REQUESTED** [review]
could you add a test for this fix please ?

**Comment by jash-odoo** [comment]
Hello @rrahir I added the test for same, can you please review this again?

File: tests/components/charts.test.ts [hunk 2]
```diff
@@ -328,12 +328,31 @@ describe("charts", () => {
...
 
+  test.each(["basicChart", "scorecard", "gauge"])(
+    "edit chart title followed by clicking on grid will update the title of %s",
+    async (chartType: string) => {
+      createTestChart(chartType);
+      await nextTick();
+      await simulateClick(".o-figure");
+      await simulateClick(".o-figure-menu-item");
+      await simulateClick(".o-menu div[data-name='edit']");
+      await nextTick();
```
Comment (rrahir) [inline tests/components/charts.test.ts:346]: you should check the current value first

File: tests/components/charts.test.ts [hunk 3]
```diff
@@ -328,12 +328,31 @@ describe("charts", () => {
...
+      await simulateClick(".o-panel .inactive");
+      setInputValueAndTrigger(".o-chart-title input", "new_title", "input");
+      setInputValueAndTrigger(".o-chart-title input", "new_title", "change");
```
Comment (rrahir) [inline tests/components/charts.test.ts:349]: In master, this specific line triggers the `updateTitle` method of the sidepanel which in turn updates the chart. This  means the test passes in master. you should remove it (which would make the test fail in master) and the see if your fix works

**Review by rrahir — CHANGES_REQUESTED** [review]

**Review by rrahir — APPROVED** [review]
Nice :)

@robodoo r+

**Pull request closed without merging**